src/main/java/frc/robot/end_effector/EndEffector.java

```diff
@@ -0,0 +1,35 @@
+package frc.robot.end_effector;
+
+import org.littletonrobotics.junction.Logger;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.end_effector.EndEffectorIO.EndEffectorIOInputs;
+
+public class EndEffector extends SubsystemBase {
+    
+    private final EndEffectorIO io;
+    private final EndEffectorIOInputs inputs = new EndEffectorIOInputs();
+
+    public EndEffector (EndEffectorIO io) {
+
+        this.io = io;
+    }
+
+   @Override
+public void periodic() {
+    io.updateInputs((EndEffectorIOInputs) inputs);
```

Comment:
You shouldn't need to cast to anything, just plug `inputs` straight into `io.updateInputs()`.